File: src/data/courses/dsa/two-pointers-sliding-window/sliding-window-variable-longest/longest-subarray-sum-k/code/python/longest_subarray_sum_k_sol1.py
def longest_subarray(nums, k):
    max_len = 0
    for i in range(len(nums)):
        s = 0
        for j in range(i, len(nums)):
            s += nums[j]
            if s <= k: max_len = max(max_len, j-i+1)
            else: break
    return max_len

File: src/data/courses/dsa/two-pointers-sliding-window/sliding-window-variable-longest/longest-subarray-sum-k/code/cpp/longest_subarray_sum_k_sol1.cpp
int longestSubarray(vector<int>& nums, int k) {
    int maxLen = 0;
    for (int i = 0; i < nums.size(); i++) {
        int sum = 0;
        for (int j = i; j < nums.size(); j++) {
            sum += nums[j];
            if (sum <= k) maxLen = max(maxLen, j-i+1);
            else break; // Sum exceeded, no point extending
        }
    }
    return maxLen;
}